### PR TITLE
Use u128::MAX in fungible ledger transfer test

### LINKED ISSUE #685: Switch to u128::MAX in fungible ledger mint integration tests
// TODO:: Switch to u128::MAX when we start using https://github.com/paritytech/substrate/pull/11241
https://github.com/Manta-Network/Manta/blob/8ab34fcb1022d4891d519bdcbb6ae22d97852ed4/runtime/calamari/tests/integration_tests.rs#L1140

### PR DESCRIPTION
**Pull request by ghzlatarev**
Signed-off-by: Georgi Zlatarev <georgi.zlatarev@manta.network>

## Description

closes: #685 

* Use u128::MAX in fungible ledger transfer test for initial balance. This case used to fail before because the `can_deposit` checks would try to increase total supply and overflow with large numbers. Fixed with https://github.com/paritytech/substrate/pull/11241
* Get rid of division by 2 of the supply in MantaPay tests which was what originally caught the bug.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Added **one** line describing your change in `<branch>/CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer.

Situational Notes:
- If adding functonality, write unit tests!
- If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.
- If runtime changes, need to update the version numbers properly:
   * `authoring_version`: The version of the authorship interface. An authoring node will not attempt to author blocks unless this is equal to its native runtime.
   * `spec_version`: The version of the runtime specification. A full node will not attempt to use its native runtime in substitute for the on-chain Wasm runtime unless all of spec_name, spec_version, and authoring_version are the same between Wasm and native.
   * `impl_version`: The version of the implementation of the specification. Nodes are free to ignore this; it serves only as an indication that the code is different; as long as the other two versions are the same then while the actual code may be different, it is nonetheless required to do the same thing. Non-consensus-breaking optimizations are about the only changes that could be made which would result in only the impl_version changing.
   * `transaction_version`: The version of the extrinsics interface. This number must be updated in the following circumstances: extrinsic parameters (number, order, or types) have been changed; extrinsics or pallets have been removed; or the pallet order in the construct_runtime! macro or extrinsic order in a pallet has been changed. You can run the `metadata_diff.yml` workflow for help. If this number is updated, then the `spec_version` must also be updated
- Verify benchmarks & weights have been updated for any modified runtime logics

### CODE REVIEW

File: pallets/manta-pay/src/test/payment.rs [hunk 1]
```diff
@@ -160,9 +160,8 @@ where
         Some(id) => id,
         None => rng.gen(),
     };
-    // FIXME: get rid of the division after parity fixes the pallet-asset bug
-    let double_balance: u128 = rng.gen();
-    let total_free_balance = AssetValue(double_balance / 2);
+    let balance: u128 = rng.gen();
```
Comment (Dengjianping) [inline pallets/manta-pay/src/test/payment.rs]: Can both line of code be a function? I mean it's being used several places.
Not sure whether it's possible.

File: runtime/calamari/tests/integration_tests.rs [hunk 2]
```diff
@@ -1108,8 +1108,7 @@ fn concrete_fungible_ledger_transfers_work() {
             ),);
 
             // Register and mint for testing.
-            // TODO:: Switch to u128::MAX when we start using https://github.com/paritytech/substrate/pull/11241
-            let amount = INITIAL_BALANCE;
+            let amount = u128::MAX;
```
Comment (Garandor) [inline runtime/calamari/tests/integration_tests.rs]: Wouldn't it be cleaner to access u128 through `manta_primitives::types::Balance::MAX`?

Comment (Garandor) [reply runtime/calamari/tests/integration_tests.rs]: Actually, this is testcode, so nvm

Comment (ghzlatarev) [reply runtime/calamari/tests/integration_tests.rs]: I can use the manta-primitives Balance, so it's more of an integration test.

File: pallets/manta-pay/src/test/payment.rs (hunk 1 above)
Comment (ghzlatarev) [reply pallets/manta-pay/src/test/payment.rs]: Just made it a 1 liner.

**Review by Garandor — DISMISSED** [review]
LGTM

**Review by Dengjianping — DISMISSED** [review]
LGTM!

**Comment by Dengjianping** [comment]
I'm afraid you have to separate a `Unreleased` section for new changelogs.

**Review by Garandor — APPROVED** [review]
LGTM

**Review by Dengjianping — APPROVED** [review]

**Pull request merged**